lld/ELF/Symbols.h

```diff
@@ -316,9 +308,25 @@ class Symbol {
   // This field is a index to the symbol's version definition.
   uint16_t verdefIndex;
 
-  // Version definition index.
-  uint16_t versionId;
+  // Used by a Defined symbol with protected or default visibility, to record
+  // the verdef index and whether the symbol is exported into .dynsym.
+  // * -1 (initial): not exported, binding unchanged
```

Comment:
Is it work saying -1u as the field is uint16_t?

I guess this means that we'll not be able to support a program with 65535 symbol versions. Probably not likely to encounter that in practice.

Reply:
The most significant bit is for VERSYM_HIDDEN 0x8000, so the largest version index is 0x7fff. This changes means that we cannot use 0x7fff as the version index. 

In practice, version indexes don't even reach 256...